### PR TITLE
Modularize recenter

### PR DESCRIPTION
**Pull request by ger-benjamin**
For GSGMF https://jira.camptocamp.com/browse/GSGMF-174

### CODE REVIEW

File: src/recenter/directive.js [hunk 1]
```diff
@@ -1,7 +1,15 @@
...
+goog.provide('ngeo.recenter.directive');
 
 goog.require('ngeo');
 
+/**
+ * @type {!angular.Module}
+ */
+ngeo.recenter.directive = angular.module('ngeoRecenter', []);
+
+const ngeoModule = ngeo.module;
```
Comment (ger-benjamin) [inline src/recenter/directive.js]: I understand why I have to create this const, but not why it was not required in the gmf.search.component...

File: src/recenter/directive.js [hunk 2]
```diff
@@ -1,7 +1,15 @@
-goog.provide('ngeo.recenterDirective');
+goog.provide('ngeo.recenter.directive');
```
Comment (gberaudo) [inline src/recenter/directive.js]: Should be in `src/map/recenter.js`. @sbrunner, do you want to add a note in https://jira.camptocamp.com/browse/GSGMF-172 explaining the files need to be renamed (not moved in a `directive` subdirectory)?

Then the symbol should be `ngeo.map.recenter`. (I would drop the `directive` suffix).

File: src/recenter/directive.js (hunk 1 above)
Comment (gberaudo) [reply src/recenter/directive.js]: @ger-benjamin, no it is not correct to use this const here: you can not create a local constant, in a `goog.provide` file.
Do you have an error when using `ngeo.module` directly?

File: src/recenter/module.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+/**
+ * @module ngeo recenter namespace
+ */
+goog.provide('ngeo.recenter.module');
```
Comment (gberaudo) [inline src/recenter/module.js]: This module is not necessary (there is no recenter top level module). You can recycle it by renaming it `ngeo.map.module` and moving it there if it does not already exist.

File: examples/recenter.js [hunk 4]
```diff
@@ -1,17 +1,20 @@
...
 /** @suppress {extraRequire} */
 goog.require('ngeo.mapDirective');
-/** @suppress {extraRequire} */
-goog.require('ngeo.recenterDirective');
 goog.require('ol.Map');
 goog.require('ol.View');
 goog.require('ol.layer.Tile');
 goog.require('ol.source.OSM');
 
+goog.require('ngeo.recenter.module');
```
[the remark this answers is not in the document]
Comment (sbrunner) [reply examples/recenter.js]: In the applications we will use in priority the tom level module, I think that's should be the same in the examples :-)

File: src/recenter/directive.js (hunk 2 above)
Comment (sbrunner) [reply src/recenter/directive.js]: +1

**Comment by ger-benjamin** [comment]
Thanks @sbrunner @gberaudo . I'll wait to merge #3206 then address your comment

**Pull request merged**